backend/btrixcloud/crawlconfigs.py

```diff
@@ -1052,6 +1080,94 @@ async def re_add_all_scheduled_cron_jobs(self):
                     flush=True,
                 )
 
+    def _validate_behavior_git_repo(self, repo_url: str, branch: str = ""):
+        """Validate git repository and branch, if specified, exist and are reachable"""
+        git_remote_cmd = ["git", "ls-remote", repo_url, "HEAD"]
+        try:
+            subprocess.run(git_remote_cmd, check=True)
+        # Raises on non-zero exit code (repo doesn't exist or isn't reachable)
+        # pylint: disable=raise-missing-from
+        except subprocess.CalledProcessError:
+            raise HTTPException(
+                status_code=404,
+                detail="custom_behavior_not_found",
+            )
+
+        if branch:
+            time.sleep(2)
```

Comment:
Maybe can reduce this? Was trying to avoid getting rate limited

```suggestion
            time.sleep(0.5)
```

Reply:
sure, that's probably ok

Reply:
Replaced with `await asyncio.sleep(0.5)`